File: observer/php/classes/EmailSubscriber.php
<?php

class EmailSubscriber implements Observer {
    private $email;

    public function __construct($email) {
        $this->email = $email;
    }

    public function update($news) {
        echo "Enviando email a {$this->email}: Noticias - {$news}\n";
    }
}

File: observer/java/classes/EmailSubscriber.java
package observer.java.classes;

import observer.java.interfaces.Observer;

public class EmailSubscriber implements Observer {
    private String email;

    public EmailSubscriber(String email) {
        this.email = email;
    }

    @Override
    public void update(String news) {
        System.out.println("Enviando email a " + email + ": Noticias - " + news);
    }
}
